services/web/server/tests/unit/isolated/test_garbage_collector_core.py

```diff
@@ -47,3 +50,31 @@ async def test_regression_remove_orphaned_services_node_ids_unhashable_type_set(
     mock_app: AsyncMock,
 ):
     await remove_orphaned_services(mock_registry, mock_app)
+
+
+async def test_regression_project_id_recovered_from_the_wrong_place(
+    faker: Faker, mocker: MockerFixture
+):
+    base_module = "simcore_service_webserver.garbage_collector_core"
+    mocker.patch(
+        f"{base_module}.is_node_id_present_in_any_project_workbench",
+        autospec=True,
+        return_value=True,
+    )
+    mocker.patch(
+        f"{base_module}.ProjectDBAPI.get_from_app_context",
+        autospec=True,
+        return_value=AsyncMock(),
+    )
+    mocker.patch(f"{base_module}.director_v2_api.stop_dynamic_service", autospec=True)
+
+    await _remove_single_service_if_orphan(
+        app=AsyncMock(),
+        interactive_service={
+            "service_host": "host",
+            "service_uuid": faker.uuid4(),
+            "user_id": 1,
+            "project_id": faker.uuid4(),
+        },
+        currently_opened_projects_node_ids={},
+    )
```

Comment:
I am not sure I understand that test.
you do not check anything after the call.

Reply:
It's a regression test and it should no fail. Previously this failed.